util/k8s/k8sApplicationService.go

```diff
@@ -67,17 +75,23 @@ type K8sApplicationService interface {
 	ApplyResources(ctx context.Context, token string, request *application.ApplyResourcesRequest, resourceRbacHandler func(token string, clusterName string, request ResourceRequestBean, casbinAction string) bool) ([]*application.ApplyResourcesResponse, error)
 	FetchConnectionStatusForCluster(k8sClientSet *kubernetes.Clientset, clusterId int) error
 	RotatePods(ctx context.Context, request *RotatePodRequest) (*RotatePodResponse, error)
+	CreatePodEphemeralContainers(req *cluster.EphemeralContainerRequest) error
+	TerminatePodEphemeralContainer(req cluster.EphemeralContainerRequest) (bool, error)
+	GetK8sServerVersion(clusterId int) (*version.Info, error)
 }
 type K8sApplicationServiceImpl struct {
-	logger                      *zap.SugaredLogger
-	clusterService              cluster.ClusterService
-	pump                        connector.Pump
-	k8sClientService            application.K8sClientService
-	helmAppService              client.HelmAppService
-	K8sUtil                     *util.K8sUtil
-	aCDAuthConfig               *util3.ACDAuthConfig
-	K8sApplicationServiceConfig *K8sApplicationServiceConfig
-	K8sResourceHistoryService   kubernetesResourceAuditLogs.K8sResourceHistoryService
+	logger                       *zap.SugaredLogger
+	clusterService               cluster.ClusterService
+	pump                         connector.Pump
+	k8sClientService             application.K8sClientService
+	helmAppService               client.HelmAppService
+	K8sUtil                      *util.K8sUtil
+	aCDAuthConfig                *util3.ACDAuthConfig
+	K8sApplicationServiceConfig  *K8sApplicationServiceConfig
+	K8sResourceHistoryService    kubernetesResourceAuditLogs.K8sResourceHistoryService
+	terminalSession              terminal.TerminalSessionHandler
+	ephemeralContainerService    cluster.EphemeralContainerService
+	ephemeralContainerRepository repository.EphemeralContainersRepository
```

Comment:
do we need this ??

Reply:
yes, i need this to find if the terminate request is for external/internal container container